core/src/main/java/org/infinispan/commands/CreateCacheCommand.java

```diff
@@ -92,9 +94,11 @@ protected void waitForCacheToStabilize(Cache<Object, Object> cache, Configuratio
       while (cacheTopology.getMembers().size() < expectedMembers || cacheTopology.getPendingCH() != null) {
          long remainingTime = timeService.remainingTime(endTime, TimeUnit.NANOSECONDS);
          try {
-            stateTransferLock.waitForTopology(cacheTopology.getTopologyId() + 1, remainingTime,
-                  TimeUnit.NANOSECONDS);
-         } catch (TimeoutException ignored) {
+            CompletableFuture<Void> topologyFuture = stateTransferLock.topologyFuture(cacheTopology.getTopologyId() + 1);
+            if (topologyFuture != null) topologyFuture.get(remainingTime, TimeUnit.NANOSECONDS);
+         } catch (ExecutionException e) {
+            throw new CacheException(e.getCause());
+         } catch (java.util.concurrent.TimeoutException e) {
```

Comment:
Sever other classes have the same code, so I'd rather keep the `waitForTopology` and `waitForTransactionData` methods in `StateTransferLock` until we've made all its callers asynchronous.